Don't warn about #nullable in disabled code:

chsienki opened the pull request:
- Check if in an active code region before warning about #nullable directives in lower lang versions
 - Add test

jcouv updated:
Fixes https://github.com/dotnet/roslyn/issues/32172

chsienki commented:
@dotnet/roslyn-compiler for review please

jcouv commented:
Fixes https://github.com/dotnet/roslyn/issues/32172

jcouv commented on src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs:
```diff
@@ -86641,5 +86641,32 @@ static void Main()
+
+            CreateCompilation(new[] { source }, options: WithNonNullTypesTrue())
+                .VerifyDiagnostics();
+        }
```
Please also test that `#nullable enable` doesn't enable the feature when it's in disabled code

chsienki commented:
Fixed for #pragma too, and added extra test to cover.

---
In reply to: [475104468](https://github.com/dotnet/roslyn/pull/34319#issuecomment-475104468) [](ancestors = 475104468)
